platform/openide.util/test/unit/src/org/openide/util/TaskTest.java

```diff
@@ -260,26 +303,32 @@ public void testWaitWithTimeOutReturnsAfterTimeOutWhenTheTaskIsNotComputedAtAll
         
         fail ("Something wrong happened the task should wait for 1000ms but it took: " + time + "\n" + log);
     }
+    
+    //--------------------------------------------------------------------------
     @Test
-    public void testWaitOnStrangeTaskThatStartsItsExecutionInOverridenWaitFinishedMethodLikeFolderInstancesDo () throws Exception {
+    public void waitOnTask_thatStartsItsExecutionWithOverridenWaitFinishedMethod() 
+            throws Exception { // like FolderInstances do
+        
         class MyTask extends Task {
-            private int values;
-            
-            public MyTask () {
-                notifyFinished ();
+
+            private int values = 0;
+
+            public MyTask() {
+                notifyFinished();
             }
-            
-            public void waitFinished () {
-                notifyRunning ();
+
+            @Override
+            public void waitFinished() {
+                notifyRunning();
                 values++;
-                notifyFinished ();
+                notifyFinished();
             }
         }
-        
-        MyTask my = new MyTask ();
-        assertTrue ("The task thinks that he is finished", my.isFinished ());
-        assertTrue ("Ok, even with timeout we got the result", my.waitFinished (1000));
-        assertEquals ("But the old waitFinished is called", 1, my.values);
+
+        MyTask my = new MyTask(); 
+        assertTrue(my.isFinished()); //The task thinks that he is finished.
```

Comment:
Please put a message so that the test report is easier to read.

Reply:
I'm confused - when the test fails just looking at this line makes is obvious what went wrong
Original messages in lines 280-282 were obviously wrong as according to https://junit.org/junit4/javadoc/4.13/org/junit/Assert.html#assertTrue(java.lang.String,%20boolean)
they get displayed only if the assertion fails.

could You propose an example?

Reply:
Opening source files is tedious if the tests were not run in the IDE. So it is useful to have some explanation in the console or test reports. Especially if many tests are failing, a quick look at the error messages help to chose on which test to focus first.

Reply:
The issue is that when the test fails you have something like : failure line 329 in the test report. Having failure line 329 "The task should have finished" gives more context.

Reply:
Is there any more generic policy concenring assertion massages I should follow or should I put this message only in this paricular case?

Reply:
More a rule of the thumb: assertEquals usually display both values which usually provides some context while assertNull / notNull/true / false don't provide such context.

Reply:
fixed in https://github.com/apache/netbeans/pull/5061/commits/4ec2cba5da4d0286cea8daef002e4e095c8e6f00